### DISCUSSION TITLE
Do not read only Snes

### DISCUSSION THREAD

**Discussion by Vek001** [sanni/cartreader#223] [Need help]
Hello.
First of all, I would like to thank sanni, the originator of this wonderful project.
and Thank you for all project People.
I respect you.

I made frist try to make sanni's cartreader.
3/10 is complete, well work it!

one is just only can not read Snes
farst, It read but "checksum Error"
I compared the normal data to the data with the error.
Almost all the values are different.
<img width="1434" alt="スクリーンショット 2021-08-15 16 46 17" src="https://user-images.githubusercontent.com/88176907/129473708-87b7d917-8d11-4e75-ae5a-e89812da3665.png">
but Now Not not read,,,

I all soldering.
Check all the pins of the SNES for turn on electricity and check the two adjacent pins for electrical short circuit.
Also I check this,this and this
![127720205-e8101685-0f6a-408a-85fe-51bd0cb92a42-1](https://user-images.githubusercontent.com/88176907/129473960-ecf28b98-a5f0-4a52-a1e3-3a82c02fe4c5.jpg)
![118635739-38461980-b7d4-11eb-86b7-ec86072a002a-1](https://user-images.githubusercontent.com/88176907/129473964-8d5e541f-ee3c-4dc5-af99-02c231f2cab1.jpg)
![118495077-1e98c980-b723-11eb-990e-e891bc836547](https://user-images.githubusercontent.com/88176907/129473967-af15cf95-7605-49ab-a4ea-40cb831ad916.jpg)

I did sanni Advice
https://github.com/sanni/cartreader/discussions/175#discussioncomment-866883

AND Next What can i check?
I have only hand electrical tester.

**Comment by sanni** [comment]
According to your hex editor screenshot, it does look like there is an error in one of the data lines. Errors in the address lines usually look less random.

We need to compare the expected value to the wrong value in binary form.

HEX		:	BINARY
right -> wrong

4C -> 0C	:	01001100 -> 00001100
41 -> 01	:	01000001 -> 00000001
53 -> 13	:	01010011 -> 00010011
58 -> 18	:	01011000 -> 00011000

In all cases bit 6 is wrong, it's 0 when it should be 1. Bit 6 is SNES D6 which is Arduino pin D31.

With an multimeter check if this trace is shorted to GND or a neighbouring pin.

![shields](https://user-images.githubusercontent.com/19169494/129479485-f7b0227e-f4b3-489b-9d76-94445b770748.jpg)

Reference:
![arduino pinout](https://user-images.githubusercontent.com/19169494/129479487-383ce70c-0287-478c-b84d-574a5d75b490.jpg)
![snes pinout](https://user-images.githubusercontent.com/19169494/129479488-e4f46f3f-1352-41f6-b64b-6023bf079b52.jpg)
![snes slot pinout](https://user-images.githubusercontent.com/19169494/129479490-c81b1a3a-a6b3-496c-8782-0baf599d4fe5.jpg)
![binary](https://user-images.githubusercontent.com/19169494/129479491-b5463a5d-d2cb-4a0a-a28d-17fea894447d.png)

**Reply by Vek001** [reply]
Once again, thanks Sunny!

I checked the pins that connect the Aruduno to the SNES.
I checked the pins that connect the Aruduno to the SNES and found no problem with the multimeterde
Then I re-soldered the D6-line.

Strangely enough, I now get checksum OK, which is very rare.
(The same location as the binary in the previous error was wrong.)
And for the first time this cartreader found the SNES of the S-1.
But it only happened once.
And I found a scratch on the base near the S-1 chip.
Maybe this is the cause?
![IMG_2969](https://user-images.githubusercontent.com/88176907/129490217-774cc529-65f8-4eeb-9100-7ff595f9e449.jpg)


HEX : I need to learn more about BINARY.
To be honest, I don't even understand what you are saying, so I will start my homework until I can understand what you are saying.

Anyway... What a Salvage our old save data from the cartridge!!
It's like a dream machine :-)

I'll be more careful next time - 3.5/10 is complete.

PLS Close THIS TOPIS------THX!

**Reply by Dakkaron** [reply]
About Hex and Binary: both systems are just different ways to display numbers, like the decimal system.

So for example, the number 14 in decimal is E in Hex or 1110 in binary. Binary is the native system for computers, but it's really hard for humans. So Hex is a  kind of middle ground. Since 16 is 2*2*2*2 it is really easy to convert from hex to binary and vice versa (especially, since 4 digits in binary always are equivalent to one digit in hex, compared to decimal, where it doesn't fit nicely).

So if you group data into bytes, each byte represents 8 bit (so 8 binary digits) or two hex digits.

The cartridges are connected in parallel to the reader. That means, in one clock cycle, 8 or 16 lanes (depending on the console) will transfer one bit each.

So if lane nr 6 is damaged, it will always read the same value, e.g. a 0. This is what happens here. If you look at the bytes, all values transferred over the 6th lane (so each bit on position 6) is 0. The bytes that are correct just so happen to already have a 0 on position 6.

Look up a hex->binary conversion table for further learning.

Also, if you use Windows, switch your calculator to programmer mode and play with Decimal/Hex/Binary numbers.

**Comment by Vek001** [comment]
Oh, what a clear explanation!
Thank you Dakkaron.

I was just asking Dr. Google about it.
But U are better than Dr. Google ;-0
I kind of understand the concept...
What made me wonder this time was that when I followed the error, I came across the GBA pins.
Is the end (beginning) of the lane GBA pins?
Strangely enough, I understand it again.

I find it very interesting.

I am a little closer to understanding again.
Thanks!
